core/app/models/spree/order.rb

```diff
@@ -516,6 +516,8 @@ def ensure_shipping_address
     end
 
     def create_proposed_shipments
+      return self.shipments if unreturned_exchange? && shipments.all?(&:shipped?)
```

Comment:
> The spec that covers why it's required is in unreturned_item_charger_spec.rb. It's a though one to bring over to order_spec.rb since it's directly related to this method being called when attempting to charge for unreturned exchanges.

I don't need it in order_spec, but nothing in that file shows that this is required. See jhawthorn/solidus@acc533053ac172550fea0bd2647ef4d91ca1234e I replace the return with a `raise "FOOBAR"` which never occurs (outside of the skipped unit test)

> I think raising might be too severe since it's not an exceptional case, perhaps wrapping the contents of create_proposed_shipments within an unless unreturned_exchange? && shipments.all?(&:shipped?) would make it a little clearer?

It is the behaviour, not the formatting that I think is unclear here. I think raising an exception is the right thing to do here: we've asked to create new shipments on an order that should not have a new shipment created. Raising an error informs the caller that we were not able to perform the requested action. Failing silently makes it impossible to determine what the result of the action was.